### PR TITLE
Generate root controller

### PR DESCRIPTION
**Pull request by shimastripe**
## WHAT

- Generate `_templates/root_controller.go.tmpl`.
- Fix the url schema in endpoint lists from the blank `""` to  `"http" or "https"`.

### CODE REVIEW

**Comment by shimastripe** [comment]
@dtan4 Sorry, the other bug still remains.

File: _templates/root_controller.go.tmpl [hunk 1]
```diff
@@ -8,7 +8,12 @@ import (
...
 
 func APIEndpoints(c *gin.Context) {
-	reqScheme := c.Request.URL.Scheme
+	var reqScheme string
+	if c.Request.TLS != nil {
+		reqScheme = "https"
+	} else {
+		reqScheme = "http"
+	}
 	reqHost := c.Request.Host
```
Comment (dtan4) [inline _templates/root_controller.go.tmpl]: Please add empty line here.

**Comment by dtan4** [comment]
Generation test failed. It seems that the template of `root_controller.go` has wrong alignment between `:` and `baseURL`. This is not your mistake, but I hope you to fix this too if you can.

```diff
 	resources := map[string]string{
-		"companies_url": baseURL + "/companies",
-		"company_url":   baseURL + "/companies/{id}",
-		"emails_url":    baseURL + "/emails",
-		"email_url":     baseURL + "/emails/{id}",
-		"jobs_url":      baseURL + "/jobs",
-		"job_url":       baseURL + "/jobs/{id}",
-		"profiles_url":  baseURL + "/profiles",
-		"profile_url":   baseURL + "/profiles/{id}",
-		"users_url":     baseURL + "/users",
-		"user_url":      baseURL + "/users/{id}",
+		"companies_url": baseURL + "/api/companies",
+		"company_url":  baseURL + "/api/companies/{id}",
+		"emails_url": baseURL + "/api/emails",
+		"email_url":  baseURL + "/api/emails/{id}",
+		"jobs_url": baseURL + "/api/jobs",
+		"job_url":  baseURL + "/api/jobs/{id}",
+		"profiles_url": baseURL + "/api/profiles",
+		"profile_url":  baseURL + "/api/profiles/{id}",
+		"users_url": baseURL + "/api/users",
+		"user_url":  baseURL + "/api/users/{id}",
 	}
```

**Comment by dtan4** [comment]
> Sorry, the other bug still remains.

Please explain or create new issue about remained bugs.

**Comment by shimastripe** [comment]
Generated endpoint resource name is incomplete.

```
"accountnames_url": "http://localhost:8080/accountnames",
```

Correct view is this.

```
"account_names_url": "http://localhost:8080/account_names",
```

This bug is minor, so I fix in this issue.

**Comment by shimastripe** [comment]
@dtan4 passed the test!

**Comment by dtan4** [comment]
LGTM 👍 Thanks ❗

File: _example/controllers/root.go [hunk 2]
```diff
@@ -8,21 +8,28 @@ import (
...
+		"company_url":   baseURL + "/api/companies/{id}",
+		"emails_url":    baseURL + "/api/emails",
+		"email_url":     baseURL + "/api/emails/{id}",
+		"jobs_url":      baseURL + "/api/jobs",
+		"job_url":       baseURL + "/api/jobs/{id}",
+		"profiles_url":  baseURL + "/api/profiles",
+		"profile_url":   baseURL + "/api/profiles/{id}",
+		"users_url":     baseURL + "/api/users",
+		"user_url":      baseURL + "/api/users/{id}",
 	}
```
Comment (dtan4) [inline _example/controllers/root.go]: このファイルって https://github.com/wantedly/apig/pull/86/files ですでに直してあるよね？

Comment (dtan4) [inline _example/controllers/root.go]: Sorry, this comment is just my misunderstanding... 🙇 plz ignore.

**Pull request merged**